### ISSUE TITLE
[1.65.0RC2] unwanted log output

### ISSUE THREAD

**Issue by remicollet** [grpc/grpc#37178]
### What version of gRPC and what language are you using?

1.65.0RC2

### What operating system (Linux, Windows,...) and version?

GNU/Linux

### What runtime / compiler are you using (e.g. python version or version of GCC)

GCC 13

### What did you do?

php --ri grpv

### What did you expect to see?

```
grpc

grpc support => enabled
grpc module version => 1.65.0RC2
...
```
### What did you see instead?
```

WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1720426999.712108  254991 config.cc:230] gRPC experiments enabled: call_status_override_on_cancellation, event_engine_dns, event_engine_listener, http2_stats_fix, monitoring_experiment, pick_first_new, trace_record_callops, work_serializer_clears_time_cache, work_serializer_dispatch
I0000 00:00:1720426999.712165  254991 ev_epoll1_linux.cc:125] grpc epoll fd: 3

grpc

grpc support => enabled
grpc module version => 1.65.0RC2
...
```
Make sure you include information that can help us debug (full error message, exception listing, stack trace, logs).

See [TROUBLESHOOTING.md](https://github.com/grpc/grpc/blob/master/TROUBLESHOOTING.md) for how to diagnose problems better.

**Comment by Tradias** [comment]
This also affects 1.64.2. Can I cherry-pick some commit to silence it?

**Comment by sreenithi** [comment]
We have currently yanked the public release version 1.65.0 until a patch is released.

**Comment by chadlwilson** [comment]
FWIW, the Ruby gems at https://rubygems.org/gems/grpc don't seem to have been formally "yanked", if that was/is the intention. But no problem; will wait for a fix!

**Comment by garyburgmann** [comment]
this is causing headaches for us in GitHub Actions re: php grpc extension

look forward to a solution!

**Comment by chadlwilson** [comment]
Looks like only yanked formally for Python/PyPi.

**Comment by sreenithi** [comment]
yes sorry my bad, that I didn't specify the language specifically. Only the Python/PyPi 1.65.0 version is yanked as of now.

**Comment by stanley-cheung** [comment]
Deleted the 1.65.0 release for PHP

**Comment by remicollet** [comment]
> Deleted the 1.65.0 release for PHP

indeed 1.65.0 was removed from https://pecl.php.net/package/gRPC
but 1.65.1 appeared and is still affected by this issue

**Comment by remicollet** [comment]
Notice: a trivial workaround is https://git.remirepo.net/cgit/rpms/php/pecl/php-pecl-grpc.git/plain/grpc-workaround.patch

**Comment by remicollet** [comment]
Instead of the previous workaround, here is a simple patch
https://git.remirepo.net/cgit/rpms/php/pecl/php-pecl-grpc.git/plain/grpc-workaround.patch

```
diff -up grpc-1.65.1/third_party/abseil-cpp/absl/log/internal/log_sink_set.cc.old grpc-1.65.1/third_party/abseil-cpp/absl/log/internal/log_sink_set.cc
--- grpc-1.65.1/third_party/abseil-cpp/absl/log/internal/log_sink_set.cc.old	2024-07-17 14:08:10.982474321 +0200
+++ grpc-1.65.1/third_party/abseil-cpp/absl/log/internal/log_sink_set.cc	2024-07-17 14:12:26.562502991 +0200
@@ -90,8 +90,7 @@ class StderrLogSink final : public LogSi
   ~StderrLogSink() override = default;
 
   void Send(const absl::LogEntry& entry) override {
-    if (entry.log_severity() < absl::StderrThreshold() &&
-        absl::log_internal::IsInitialized()) {
+    if (entry.log_severity() < absl::StderrThreshold()) {
       return;
     }
 

```

**Comment by remicollet** [comment]
See https://github.com/abseil/abseil-cpp/pull/1718

**Comment by chadlwilson** [comment]
I do note this problem is somewhat called out at https://github.com/grpc/proposal/blob/master/L117-core-replace-gpr-logging-with-abseil-logging.md#action-items (and https://github.com/abseil/abseil-cpp/issues/1656 ) however there is nothing in the release notes for 1.65 I can see telling users that they are going to need to initialize Abseil themselves to avoid spurious logging, or how to go about doing it via various language bindings so I am a bit perplexed as to what users (especially non C++) are supposed to do.

There is still cruft coming into logs in some areas for Ruby with `1.65.1` as well.

**Comment by gibsondan** [comment]
Just echoing that while the volume has been reduced, 1.65.1 still has quite a bit of logspew that was not present in 1.64.1, like the following: https://gist.github.com/gibsondan/eb381bfe29371cfee5f25ce914d15662

**Comment by gibsondan** [comment]
A way to reproduce the logspew above is to following the quickstart instructions for our project here, which spins up a grpc server locally as a subprocess: https://docs.dagster.io/getting-started/quickstart#option-1-running-locally

**Comment by gracevivi523** [comment]
Still seeing the excessive logs with 1.65.1 for python projects using python3.10

**Comment by chr-frick** [comment]
> Still seeing the excessive logs with 1.65.1 for python projects.

same problem, we have downgraded to 1.64.1. the problem doesn't exist there

**Comment by sebastian-correa** [comment]
Can confirm that we still see excess logs in `1.65.1` with Python `3.10` on MacOS with M chips (not sure if that's relevant)

```python
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1721927346.615380 27743118 config.cc:230] gRPC experiments enabled: call_status_override_on_cancellation, event_engine_dns, event_engine_listener, http2_stats_fix, monitoring_experiment, pick_first_new, trace_record_callops, work_serializer_clears_time_cache
I0000 00:00:1721927346.620056 27743118 check_gcp_environment_no_op.cc:29] ALTS: Platforms other than Linux and Windows are not supported
I0000 00:00:1721927365.383239 27743118 check_gcp_environment_no_op.cc:29] ALTS: Platforms other than Linux and Windows are not supported
I0000 00:00:1721927366.870585 27743118 check_gcp_environment_no_op.cc:29] ALTS: Platforms other than Linux and Windows are not supported
I0000 00:00:1721927394.218088 27743118 check_gcp_environment_no_op.cc:29] ALTS: Platforms other than Linux and Windows are not supported
```

Happy to provide more detail if needed.

**Comment by sreenithi** [comment]
Acknowledged. Looks like the patch in 1.65.1 didn't fix it. We're working on fixing this issue currently and will upload a new patch soon.

**Comment by casperisfine** [comment]
The Ruby version also sends tons of logs on STDERR. But in addition it seems that it broke the forking support, as among these unwanted logs there is:

```
Other threads are currently calling into gRPC, skipping fork() handlers
I0000 00:00:1721804934.995408     348 fork_posix.cc:77] Other threads are currently calling into gRPC, skipping fork() handlers
I0000 00:00:1721804935.198725     348 fork_posix.cc:77] Other threads are currently calling into gRPC, skipping fork() handlers
```

Which are confirmed is incorrect. I don't have a specific repro, but I was able to inspect all the running (Ruby level) threads and none were calling into GRPC.

**Comment by chadlwilson** [comment]
> The Ruby version also sends tons of logs on STDERR. But in addition it seems that it broke the forking support, as among these unwanted logs there is:
> 
> ```
> Other threads are currently calling into gRPC, skipping fork() handlers
> I0000 00:00:1721804934.995408     348 fork_posix.cc:77] Other threads are currently calling into gRPC, skipping fork() handlers
> I0000 00:00:1721804935.198725     348 fork_posix.cc:77] Other threads are currently calling into gRPC, skipping fork() handlers
> ```
> 
> Which are confirmed is incorrect. I don't have a specific repro, but I was able to inspect all the running (Ruby level) threads and none were calling into GRPC.

I have also had a problem with forking or something similar with 1.65.0 in ruby but have not figured out the root cause or whether log related so was waiting for this to be cleaned up and the dust settles before digging deeper.

**Comment by sourabhsinghs** [comment]
@chongshenng  whats the log load you are experiencing and at what rate ?

**Comment by chongshenng** [comment]
@sourabhsinghs The log load is minimal and that is the output as a result of starting a gRPC server. If we downgrade to `1.64.3`, we don't see this output. 

Happy to provide more details if needed.

**Comment by vvoicehovics** [comment]
was having same issue in php

`I0000 00:00:1723113215.110756       1 call_credentials.c:168] GRPC_PHP: call credentials plugin function - begin`

grpc support => enabled
grpc module version => 1.65.2

was able to fix with 
grpc.grpc_verbosity = ERROR

**Comment by Tradias** [comment]
I can confirm no more unwanted log output in 1.65.4 (compared to 1.65.2) when using gRPC from C++.

**Comment by Olegt0rr** [comment]
Just checked `1.65.4` - unwanted log output is gone

It seems the issue should be closed

**Comment by chongshenng** [comment]
@Olegt0rr we still see [these logs](https://github.com/grpc/grpc/issues/37178#issuecomment-2271374448) on version `1.65.4`. For context, I tested this on MacOS 14.5 with Python `3.11.8` and `3.10.9`.

_However_, I can confirm we do not see these addition logs on version `1.65.4` when running on Ubuntu 22.04.4 with Python `3.11.8` and `3.10.9`. Maybe this is useful @sourabhsinghs?

**Comment by Olegt0rr** [comment]
Looks like logs from issue and logs from other comments should be in different issue. Other way it would never be closed :D

**Comment by Taucher2003** [comment]
`1.65.4` has not been published on https://rubygems.org/gems/grpc/versions

Newest version there is `1.65.2` (excluding the pre-release versions) which includes this unwanted log output.

Any chance to get this version published for Ruby?

**Comment by Linuus** [comment]
Yeah it would be really great to get the 1.65.4 release in Rubygems to fix the messy log output.

**Comment by apolcyn** [comment]
Sorry for delay, the ruby 1.66.0 release removes *most* log spam.

There is still one log spam line affecting ruby servers which I've opened https://github.com/grpc/grpc/pull/37633 for. After that is in, I plan to backport that to 1.66

Separately I plan to backport https://github.com/grpc/grpc/pull/37634  and https://github.com/grpc/grpc/pull/37633 (as soon as it's merged) to 1.65

**Comment by jeremylaier-tc** [comment]
This issue is still happening on macos. grpcio v1.65.4

**Comment by sreenithi** [comment]
@jeremylaier-tc can you clarify if the log load you are seeing is similar to that [noted by chongshenng@ above](https://github.com/grpc/grpc/issues/37178#issuecomment-2277623774) or if it's different?

**Comment by chongshenng** [comment]
@jeremylaier-tc @sreenithi FYI, I've opened #37642 to track the MacOS issue (but for `1.66.1`. If the same unwanted logs are also showing in the MWE for `1.65.4`, we could update the issue to also include that version? @jeremylaier-tc wdyt?
